src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -85,12 +113,41 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
 
                     _reindexJobRecord.Resources.AddRange(resourceList);
                     _reindexJobRecord.SearchParams.AddRange(notYetIndexedParams.Select(p => p.Name));
-                }
 
-                // This is just a shell for now, will be completed in future
-                await CompleteJobAsync(OperationStatus.Completed, cancellationToken);
+                    // generate and run first query
+                    var queryStatus = new ReindexJobQueryStatus(null);
+                    queryStatus.LastModified = DateTimeOffset.UtcNow;
+                    queryStatus.Status = OperationStatus.Queued;
+
+                    _reindexJobRecord.QueryList.Add(queryStatus);
+
+                    // update the complete total
+                    var countOnlyResults = await ExecuteReindexQueryAsync(queryStatus, countOnly: true, cancellationToken);
+                    _reindexJobRecord.Count = countOnlyResults.TotalCount.Value;
+
+                    // Query first batch of resources
+                    await ProcessQueryAsync(queryStatus, cancellationToken);
+                }
+                else
+                {
+                    // check to see if queries are queued
+                    // TODO: this while loop is temporary until we multithread this task so multiple threads can
+                    // processes queries
+                    while (_reindexJobRecord.QueryList.Where(q => q.Status == OperationStatus.Queued).Any())
+                    {
+                        // grab the next query from the list which is labeled as queued and run it
+                        var query = _reindexJobRecord.QueryList.Where(q => q.Status == OperationStatus.Queued).OrderBy(q => q.LastModified).FirstOrDefault();
 
-                _logger.LogTrace("Successfully completed the job.");
+                        if (query != null)
```

Comment:
Why is the if statement needed? It looks like the while loop would require query to not be null.

Reply:
You are correct.  I removed it.